src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -61,6 +61,7 @@
 public class GatewayAllocator extends AbstractComponent {
 
     public static final String INDEX_RECOVERY_INITIAL_SHARDS = "index.recovery.initial_shards";
+    public static final String INDEX_ALLOW_RECOVERY_ON_ANY_NODE = "index.recovery.recover_on_any_node";
```

Comment:
can we call this `index.shared_filesystem.recover_on_any_node` (and rename the constant) and move it to IndexMetaData? I am concerned that people will think its a flag that supports non shared file system cases